mllib/src/main/scala/org/apache/spark/mllib/api/python/PythonMLLibAPI.scala

```diff
@@ -985,6 +985,7 @@ private[spark] object SerDe extends Serializable {
       val m: DenseMatrix = obj.asInstanceOf[DenseMatrix]
       val bytes = new Array[Byte](8 * m.values.size)
       val order = ByteOrder.nativeOrder()
+      val isTransposed = if (m.isTransposed) 1 else 0
       ByteBuffer.wrap(bytes).order(order).asDoubleBuffer().put(m.values)
 
       out.write(Opcodes.BININT)
```

Comment:
Before this line, put `out.write(Opcodes.MARK)`, which marks the start of the tuple.